Decode/EventHeaderEnumerator.cs

```diff
@@ -1092,19 +1092,23 @@ public EventHeaderEventInfo GetEventInfo(ReadOnlySpan<byte> eventDataSpan)
         /// </para>
         /// </summary>
         /// <exception cref="InvalidOperationException">Called in invalid State.</exception>
-        public PerfItemType GetItemType()
+        public PerfItemMetadata GetItemMetadata()
         {
             if (m_state <= EventHeaderEnumeratorState.BeforeFirstItem)
             {
                 throw new InvalidOperationException(); // PRECONDITION
             }
 
-            return new PerfItemType(
+            var isScalar =
+                m_state < EventHeaderEnumeratorState.ArrayBegin ||
+                m_state > EventHeaderEnumeratorState.ArrayEnd;
+            return new PerfItemMetadata(
                 m_byteReader,
                 m_fieldType.Encoding,
                 m_fieldType.Format,
+                isScalar,
```

Comment:
Note: It was really inconvenient and confusing that the PerfItemMetadata class (previously PerfItemType) did not have `IsElement` and `IsScalar` properties:

- `IsScalar`: true if this is a non-array item, or a single element of an array field; false if this is an array-begin or array-end item.
- `IsElement`: true if this is a single element of an array field; false if this is a non-array item, array-begin, or array-end.

Fix: PerfItemMetadata now directly tracks `IsScalar`, and from this it can indirectly determine `IsElement = IsScalar && 0 != ArrayFlag`.